pkg/mcs/tso/server/server.go

```diff
@@ -517,32 +519,39 @@ func (s *Server) startGRPCAndHTTPServers(l net.Listener) {
 	go s.startGRPCServer(grpcL)
 	go s.startHTTPServer(httpL)
 
-	if err := mux.Serve(); !strings.Contains(err.Error(), "use of closed network connection") {
-		log.Fatal("mux stop serving unexpectedly.", errs.ZapError(err))
+	if err := mux.Serve(); err != nil {
+		if s.IsClosed() {
+			log.Info("mux stop serving.", errs.ZapError(err))
+		} else {
+			log.Fatal("mux stop serving unexpectedly.", errs.ZapError(err))
+		}
 	}
 }
 
-func (s *Server) startServer(ctx context.Context) (err error) {
+func (s *Server) startServer() (err error) {
 	if s.clusterID, err = etcdutil.InitClusterID(s.etcdClient, tsoClusterIDPath); err != nil {
 		return err
 	}
 	log.Info("init cluster id", zap.Uint64("cluster-id", s.clusterID))
+
+	s.serverLoopCtx, s.serverLoopCancel = context.WithCancel(s.ctx)
```

Comment:
where do we use? I think we can remove them

Reply:
removed s.serverLoopCtx, s.serverLoopCancel = context.WithCancel(s.ctx) to startServerLoop()